src/libexpr/primops/fetchTree.cc

```diff
@@ -529,9 +529,20 @@ static void prim_fetchurl(EvalState & state, const PosIdx pos, Value * * args, V
 
 static RegisterPrimOp primop_fetchurl({
     .name = "__fetchurl",
-    .args = {"url"},
+    .args = {"args"},
     .doc = R"(
-      Download the specified URL and return the path of the downloaded file.
+      If args is a URL, return the path of the downloaded file.
+      Otherwise, it can be an attribute with the following attributes
+      (all except url are optional):
+
+      - `url`
+
+        The URL of the file to download.
+
+      - `name` (default: `url without the protocol`)
```

Comment:
```suggestion
      - `name` (default: the last path component of the URL)
```
or something like that.